integration-core/src/main/java/org/symphonyoss/integration/core/service/CryptoServiceImpl.java

```diff
@@ -92,31 +71,20 @@
   public String encrypt(String plainText, String key) throws CryptoException {
     checkParameters("plainText", plainText);
     checkParameters("key", key);
-    try {
-      byte[] saltBytes = generateSalt();
-      SecretKeySpec secret = deriveKey(key, saltBytes);
-
-      Cipher cipher = getCipher();
-      initCipher(cipher, Cipher.ENCRYPT_MODE, secret, null);
-      AlgorithmParameters params = cipher.getParameters();
-      byte[] ivBytes = params.getParameterSpec(IvParameterSpec.class).getIV();
-      byte[] encryptedTextBytes = doCipher(cipher, plainText.getBytes(CHARSET));
-
-      // Prepend Salt and VI
-      byte[] buffer = new byte[saltBytes.length + ivBytes.length + encryptedTextBytes.length];
-      System.arraycopy(saltBytes, 0, buffer, 0, saltBytes.length);
-      System.arraycopy(ivBytes, 0, buffer, saltBytes.length, ivBytes.length);
-      System.arraycopy(encryptedTextBytes, 0, buffer, saltBytes.length + ivBytes.length,
-          encryptedTextBytes.length);
-
-      return Base64.encodeBase64String(buffer);
-    } catch (InvalidParameterSpecException e) {
-      throw new CryptoException(logMessage.getMessage(INVALID_PARAM_SPEC), e,
-          logMessage.getMessage(INVALID_PARAM_SPEC_SOLUTION));
-    } catch (UnsupportedEncodingException e) {
-      throw new CryptoException(logMessage.getMessage(UNSUPPORTED_ENCODING), e,
-          logMessage.getMessage(UNSUPPORTED_ENCODING_SOLUTION, CHARSET));
-    }
+
+    // Generate a random salt
+    SecureRandom random = new SecureRandom();
```

Comment:
You can create this object as a class attribute once SecureRandom class must produce non-deterministic output on each call. What that means is creating a single instance does not affect the randomness of the random bytes it generates. Thus, you can create just one instance rather than too many instances.
It can be helpful if you need to mock something on this object.

Reply:
You can reuse what's already in the crypto lib. : Bytes.random(). This way there is no need to deal with SecureRandom directly.

Reply:
Done. Using Bytes.random(). Since it's static, there'll be no attribute for this object anymore.